lib/src/test/kotlin/org/didcommx/didcomm/pack/signed/PackSignedTest.kt

```diff
@@ -0,0 +1,87 @@
+package org.didcommx.didcomm.pack.signed
+
+import org.didcommx.didcomm.DIDComm
+import org.didcommx.didcomm.Person
+import org.didcommx.didcomm.crypto.key.RecipientKeySelector
+import org.didcommx.didcomm.fixtures.JWM
+import org.didcommx.didcomm.getAuthMethodsInSecrets
+import org.didcommx.didcomm.message.Message
+import org.didcommx.didcomm.messages.attachmentJsonMsg
+import org.didcommx.didcomm.messages.attachmentMulti1msg
+import org.didcommx.didcomm.mock.AliceSecretResolverMock
+import org.didcommx.didcomm.mock.BobSecretResolverMock
+import org.didcommx.didcomm.mock.DIDDocResolverMock
+import org.didcommx.didcomm.model.PackSignedParams
+import org.didcommx.didcomm.model.UnpackParams
+import org.didcommx.didcomm.operations.unpack
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.stream.Stream
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+data class PackSignedTestData(
+    val msgList: List<Message>,
+    val signedFromList: List<String>
+)
+
+class PackSignedTest {
+    companion object {
+
+        @JvmStatic
+        fun packSignedTest(): Stream<PackSignedTestData> {
+            val signedFromList = getAuthMethodsInSecrets(Person.ALICE).map { it.id }.toMutableList()
+            signedFromList.add(JWM.ALICE_DID)
+
+            return Stream.of(
+                PackSignedTestData(
+                    listOf(JWM.PLAINTEXT_MESSAGE, attachmentMulti1msg(), attachmentJsonMsg()),
+                    signedFromList
+                ),
+            )
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("packSignedTest")
+    fun testAnoncrypt(data: PackSignedTestData) {
```

Comment:
ParameterizedTest doesn't make sense here as we always return just one element in the stream.
You can choose either to remove the parameter function and define the expected lists of parameters and for loops explicitly in the test, or try to do the following:

```
data class PackSignedTestData(
    val msg: Message,
    val signedFrom: String
)
```
```
       @JvmStatic
        fun packSignedParams(): Stream<PackSignedTestData> {
            val signedFromList = getAuthMethodsInSecrets(Person.ALICE).map { it.id }.toMutableList()
            signedFromList.add(JWM.ALICE_DID)
            val msgs = listOf(JWM.PLAINTEXT_MESSAGE, attachmentMulti1msg(), attachmentJsonMsg())
            val cartesianProduct = .... 
            return Stream.of(cartesianProduct)
        }
```